app/Domains/Donors/Actions/Queries/GetDonors.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domains\Donors\Actions\Queries;
+
+use App\Domains\Donors\Donor;
+use Illuminate\Auth\Access\Response;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
+use Lorisleiva\Actions\ActionRequest;
+use Lorisleiva\Actions\Concerns\AsAction;
+
+class GetDonors
+{
+    use AsAction;
+
+    public function authorize(ActionRequest $request): Response
+    {
+        $user = $request->user();
+        
+        if ($user->has_general_access) {
+            return Response::allow();
+        }
+
+        return Response::deny('You are unauthorized to perform this action');
+    }
+
+    public function handle(
+        ?int $per_page = 10,
+         ?string $search_term = ''
+    ): Collection | LengthAwarePaginator {
+        $query = Donor::query();
+
+        if ($search_term) {
+            $query
+                ->where('name', 'like', "%$search_term%")
+                ->orWhere('details', 'like', "%$search_term%")
+                ->orWhere('phone', 'like', '%' . $search_term . '%')
+                ->orWhere('email', 'like', '%' . $search_term . '%')
+                ->orWhere('address', 'like', '%' . $search_term . '%')
+                ->orWhere('details', 'like', '%' . $search_term . '%');
+        }
+    
+        return $per_page === null ?
+            $query->get() :
+            $query->paginate($per_page);
+    }
+
+    public function rules(): array
+    {
+        return [
+            'search_term' => ['nullable', 'string'],
+            'per_page' => ['nullable', 'integer', 'min:1'],
+        ];
+    }
+
+    public function asController(ActionRequest $request)
+    {
+        $search_term = $request->input('search_term');
+        $per_page = $request->input('per_page');
+
+        return $this->handle($per_page, $search_term);
```

Comment:
This might be a bit better:
```php
return $this->handle(
    $request->input('per_page'),
    $request->input('search_term'),
)
```

The reason for that is because additional memory does not have to be allocated to store the variables `$per_page` and `$search_term`.

Also, you are not validating the request input here, the `rules` function is pretty much useless here.